### DISCUSSION TITLE
Rust Doc upload Via SSH, SCP or RSYNC

### DISCUSSION THREAD

**Discussion by tomus85** [kellnr/kellnr#336] [Q&A]
Currently, I am having trouble with the upload doc command. Raised the issue here #287.

I was thinking about this. It might be possible for me to upload the doc with one of the commands: `SSH`, `SCP` or `RYNC`.

I have had a look at the `data` directory used for the docker and noticed the following dirs: `crates`, `doc_queues` and `docs` So I was thinking of uploading the crate doc to `doc`. I want to know if there is a specific folder structure for this. 

Looking at `crates` the structure is in this format: `crate_name-version.crate` example: `my_helper-0.1.0.crate` 

I am hoping I can simply upload the doc dir to remote (kellnr docker -> data dir) 

If someone can help me out with the specific file / folder structure then hopefully this should work

Cheers
Tom

**Comment by secana** [comment]
The directory is `<data_dir>/docs/<package>/<version>`

For example:

Crate: foo-bar
Version: 3.4.1
Kellnr data directory: /opt/kdata

The resulting directory for the docs is; `/opt/kdata/docs/foo-bar/3.4.1/doc`
This folder should contain the *.html and *.js files generated by `cargo doc`.

I hope that helps.

**Reply by tomus85** [reply]
Hey,

Thanks for the reply.

I have created a shell script to upload the doc onto the remote server.

I have created the following dir: `<Home>/docker/data/docks/<create_name>/<create_version>/doc` In the `/doc` this contains all the `*.html/*.js` generated by the `cargo doc` command. 

However, when viewing the created project via the web interface, on the right-hand side under `Documentation` it says "Add" I've tried refreshing the page to see if there would be a link to the crate doc. 

Now I have the doc uploaded on the server (hopefully the right dir) but how to I view this under the web interface? 

Tom

**Comment by tomus85** [comment]
I have managed to update the docker image to 5.2.3 and managed to upload the doc. Thank you for that
